packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts

```diff
@@ -0,0 +1,124 @@
+		sorted[ key ] = o[ key ];
+		return sorted;
+	}, {} as any ) as FeatureKey;
+
+	return JSON.stringify( sorted );
+}
+
+/**
+ * Turns an array of `FeastureCriteria` into a Map that can be easily queried
+ * with stringified `FeatureKey`s.
+ *
+ * @param {FeatureCriteria[]} criteria An array of `FeatureCriteria` objects.
+ * @param {FeatureMap} map Instance of `FeatureMap` that will hold the table  queriable by
+ * (stringified)`FeatureKey`s objects.
+ * @returns {FeatureMap} A `Map`object built from the `criteria` that can be queried by a
+ * (stringified) `FeatureKey` object.
+ */
+function criteriaToMap( criteria: FeatureCriteria[], map: FeatureMap ): FeatureMap {
+	return criteria.reduce( ( acc, criteria ) => {
+		const { accountName, ...rest } = criteria;
+		acc.set( stringifyKey( rest ), accountName );
+		return acc;
+	}, map );
+}
+
+// NOTE If this gets too big, move to its own dedicated module
+const defaultCriteria: FeatureCriteria[] = [
+	{
+		gutenberg: 'edge',
+		siteType: 'simple',
+		accountName: 'gutenbergSimpleSiteEdgeUser',
+	},
+	{ gutenberg: 'stable', siteType: 'simple', accountName: 'gutenbergSimpleSiteUser' },
+	// The CoBlocks account name takes precedence if CoBlocks edge
+	// is present. We have two definitions below to effectivelly
+	// ignore gutenberg in this case:
+	{
+		coblocks: 'edge',
+		gutenberg: 'stable',
+		siteType: 'simple',
+		accountName: 'coBlocksSimpleSiteEdgeUser',
+	},
+	{
+		coblocks: 'edge',
+		gutenberg: 'edge',
+		siteType: 'simple',
+		accountName: 'coBlocksSimpleSiteEdgeUser',
+	},
+	{
+		gutenberg: 'stable',
+		siteType: 'simple',
+		variant: 'siteEditor',
+		accountName: 'siteEditorSimpleSiteUser',
+	},
+	{
+		gutenberg: 'edge',
+		siteType: 'simple',
+		variant: 'siteEditor',
+		accountName: 'siteEditorSimpleSiteEdgeUser',
+	},
+];
+
+const defaultAccountsTable = criteriaToMap( defaultCriteria, new Map() );
+
+/**
+ * Return a WPCOM account name that can be passed over to build a `TestAccount`
+ * instance for a test. The account name returned will depend on the attributes
+ * passed as part of the `feature` param. The table of criteria for each account
+ * can be found in the `defaultCriteria` const that lives in the same module where
+ * this function is defined.
+ *
+ * @param {FeatureKey} feature represents a certain feature that has an account
+ * associated with in the criteria table. It will be used as a key to get the
+ * right account name.
+ * @param {FeatureCriteria[]} mergeAndOverride Can be used to pass a custom table that will
+ * be merged into the default one. Useful to do one-off criteria->account overrides for
+ * specifis tests inline. The entries passed here will replace any matched (by key) entries
+ * in the default table.
+ * @returns {string} the account name that can be used to build a new `TestAccount` instance.
+ */
+export function getTestAccountByFeature(
+	feature: FeatureKey,
+	mergeAndOverride?: FeatureCriteria[]
```

Comment:
Again, a tiny style nit here, but this name sounds kind of `boolean` to me, could we make it `mergeAndOverrideCriteria`? I know that's long, but might make a little more sense to consumers.

Reply:
Sounds good! 👍🏻